drivers/usb/device/usb_dc_mcux.c

```diff
@@ -68,7 +71,35 @@ static void usb_isr_handler(void);
 #define EP_ABS_IDX(ep)		(USB_EP_GET_IDX(ep) * 2 + \
 					(USB_EP_GET_DIR(ep) >> 7))
 #define NUM_OF_EP_MAX		(DT_INST_PROP(0, num_bidir_endpoints) * 2)
-#define CONTROLLER_ID		(DT_INST_ENUM_IDX(0, usb_controller_index))
+
+#define COUNT_INST(n) 1 +
+#define NUM_INSTS (DT_INST_FOREACH_STATUS_OKAY(COUNT_INST) 0)
```

Comment:
Why not use `DT_NUM_INST_STATUS_OKAY(compat)`?

Reply:
because my brain did not house this knowledge until now, thanks